packages/cli-kit/src/analytics.ts

```diff
@@ -108,3 +108,20 @@ const buildPayload = async (errorMessage: string | undefined, currentTime: numbe
     },
   }
 }
+
+export type ProjectType = 'node' | 'php' | 'ruby' | undefined
+
+export async function getProjectType(directory: string): Promise<ProjectType> {
```

Comment:
If this is only for internal use, I wouldn't export it and change the tests to check the `project_type` reported for every case.

Otherwise, I would move it to another file, it feels weird to import `getProjectType` from an analytics module, don't you think?

Reply:
Good idea. I've [made ](https://github.com/Shopify/cli/pull/111/commits/5457538054e7d0a420362231b471502df28f4b4a#diff-04dd7360e7e77135b59ccb6c6dc9bf5405b8003d0e93785fb2188c007b2ea186R60)the function internal and updated the `reportEvent` tests to test all the project type scenarios.